modelon/impact/client/entities/workspace.py

```diff
@@ -446,6 +451,85 @@ def created_at(self) -> int:
         return self._data["createdAt"]
 
 
+class ReceivedFromWorkspace:
+    def __init__(self, workspace: Workspace) -> None:
+        self._workspace = workspace
+
+    def get_workspace(self) -> Optional[PublishedWorkspace]:
+        """Return the published workspace from which the workspace was imported from.
+        Returns None if workspace is not linked to any published workspace.
+
+        Returns:
+            An PublishedWorkspace class object.
+
+        Example::
+
+            published_ws = workspace.received_from.get_workspace()
+
+        """
+        recieved_from = self._workspace.definition.received_from
+        if not recieved_from:
+            raise NoAssociatedPublishedWorkspaceError(
+                "No published workspace found that are "
+                f"associated with local workspace with ID: {self._workspace.id}"
+            )
```

Comment:
Seems more logical to have this check before the object gets created ( in the constructor) and not here.

I think a shorter message is sufficient: "The workspace has no link to a published workspace."